crates/types/src/net/partition_processor_manager.rs

```diff
@@ -74,7 +75,13 @@ pub struct CreateSnapshotRequest {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct CreateSnapshotResponse {
-    pub result: Result<SnapshotId, SnapshotError>,
+    pub result: Result<Snapshot, SnapshotError>,
```

Comment:
Is this a reasonable shape for this RPC response? I made it this way when I first introduced it thinking it would be more evolvable - is that actually true? Should this just be a type alias for `Result<Snapshot, SnapshotError>` rather?

Reply:
Compared to the previous version, that's better, although in the future we won't allow for marshaling "Result" like this. Since it has been already like this, I don't mind the change you introduced.

Reply:
Ah, sorry about that! I remember I called attention to this during the initial PR but didn't get clarity on this.